saleor/graphql/product/mutations/products.py

```diff
@@ -270,7 +280,14 @@ class Meta:
 
     @classmethod
     def perform_mutation(cls, _root, info, **kwargs):
+        node_id = kwargs.get("id")
+
+        instance = cls.get_node_or_error(info, node_id, only_type=Collection)
+        products = list(instance.products.all())
```

Comment:
Do you need to convert them to list?
What about prefetches?  variant, categories, collections,  attributes(?)

Reply:
I converted this into list because, this is delete mutation. Without casting to a list, after deleting collection, `instance` will doesn't had `products`.

I will add prefetch.

Reply:
> I converted this into list because, this is delete mutation. Without casting to a list, after deleting collection, `instance` will doesn't had `products`.

My bad, I missed the name of the mutation.